DatadogSessionReplay/Tests/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/WKWebViewRecorderTests.swift

```diff
@@ -73,12 +74,30 @@ class WKWebViewRecorderTests: XCTestCase {
             return XCTFail("First wireframe needs to be webviewWireframe case")
         }
 
-        XCTAssertEqual(wireframe.id, id)
-        XCTAssertEqual(wireframe.slotId, String(slotId))
+        XCTAssertEqual(wireframe.id, Int64(webView.hash))
+        XCTAssertEqual(wireframe.slotId, String(webView.hash))
         XCTAssertNil(wireframe.clip)
         XCTAssertEqual(wireframe.x, Int64(withNoOverflow: attributes.frame.minX))
         XCTAssertEqual(wireframe.y, Int64(withNoOverflow: attributes.frame.minY))
         XCTAssertEqual(wireframe.width, Int64(withNoOverflow: attributes.frame.width))
         XCTAssertEqual(wireframe.height, Int64(withNoOverflow: attributes.frame.height))
+        XCTAssertTrue(wireframe.isVisible ?? false)
+        XCTAssertTrue(builder.hiddenWebViewWireframes().isEmpty, "webview slot should be removed from builder")
     }
+
+// This test should pass but it doesn't because `WKWebView` apparently leaks.
```

Comment:
Can we confirm the behaviour in the real usage?

Reply:
The commented code was showcasing the leak, webkit is probably holding the reference. Now we rely on `NSHashTable` which states:
> The major option is to provide for "weak" references that are removed automatically, but at some indefinite point in the future.

Which is completely acceptable for our use case.